db/migrate/20220824172503_create_timeline_events.rb

```diff
@@ -0,0 +1,13 @@
+class CreateTimelineEvents < ActiveRecord::Migration[7.0]
+  def change
+    create_table :timeline_events do |t|
+      t.string :event_type, null: false
+      t.references :application_form, foreign_key: true
```

Comment:
```suggestion
      t.references :application_form, null: false, foreign_key: true
```

Sadly the default is `null: true`.